lib/openstudio-standards/standards/Standards.Model.rb

```diff
@@ -2622,6 +2622,31 @@ def model_find_objects(hash_of_objects, search_criteria, capacity = nil, date =
       end
     end
 
+    # If volume was specified, narrow down the matching objects
+    unless volume.nil?
+      # Skip objects that don't have fields for minimum_storage and maximum_storage
+      matching_objects = matching_objects.reject { |object| !object.key?('minimum_storage') || !object.key?('maximum_storage') }
+
+      # Skip objects that don't have values specified for minimum_storage and maximum_storage
+      matching_objects = matching_objects.reject { |object| object['minimum_storage'].nil? || object['maximum_storage'].nil? }
+
+      # Round up if volume is an integer
+      if volume == volume.round
+        volume += (volume * 0.01)
+      end
+      # Skip objects whose the minimum volume is below or maximum volume above the specified volume
+      matching_volume_objects = matching_objects.reject { |object| volume.to_f <= object['minimum_storage'].to_f || volume.to_f > object['maximum_storage'].to_f }
+
+      # If no object was found, round the volume down in case the number fell between the limits in the json file.
+      if matching_volume_objects.size.zero?
+        volume *= 0.99
+        # Skip objects whose minimum volume is below or maximum volume above the specified volume
+        matching_objects = matching_objects.reject { |object| volume.to_f <= object['minimum_storage'].to_f || volume.to_f > object['maximum_storage'].to_f }
```

Comment:
Again, I think this code is not necessary. The capacity has it between there is 0.01 gap between the prev record maximum and next record's minimum. The water heater storage data does not have such pattern and adding this code could cause error in searching.